app/services/optout/OptOutService.scala

```diff
@@ -128,13 +156,26 @@ class OptOutService @Inject()(itsaStatusUpdateConnector: ITSAStatusUpdateConnect
     }
   }
 
-  def saveIntent(intent: TaxYear)(implicit hc: HeaderCarrier): Future[Boolean] = {
-    val data = UIJourneySessionData(
-      sessionId = hc.sessionId.get.value,
-      journeyType = OptOutJourney.Name,
-      optOutSessionData = Some(OptOutSessionData(selectedOptOutYear = Some(intent.toString)))
-    )
-    repository.set(data)
+  def saveIntent(intent: TaxYear)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Boolean] = {
+    OptionT(repository.get(hc.sessionId.get.value, OptOutJourney.Name)).
```

Comment:
let's try and simplify to imporve readability

Reply:
may be like this
![Screenshot 2024-07-12 at 15 00 15](https://github.com/user-attachments/assets/497b1767-e0f3-4005-aee8-309175629e85)